### DISCUSSION TITLE
Adding default Metric Tags to all Metrics

### DISCUSSION THREAD

**Discussion by JackRhodes** [open-telemetry/opentelemetry-dotnet#5669] [Q&A] [answered]
I've been trying to find a way to add default tags to all metrics.

I can see there is a `tags` parameter in the Meter, so have tried to use that. However, the values do not seem to be showing

Example code:

```
// See https://aka.ms/new-console-template for more information

using System.Diagnostics.Metrics;
using OpenTelemetry;
using OpenTelemetry.Metrics;
using OpenTelemetry.Resources;

Console.WriteLine("Hello, World!");
using var meter = new Meter("TestMeter", "1.0.0", new[]
{
    new KeyValuePair<string, object?>("namespace", "jacklocaldev")
});

var providerBuilder = Sdk.CreateMeterProviderBuilder()
    .ConfigureResource(r => r.AddService("testservice"))
    .AddMeter(meter.Name); // All instruments from this meter are enabled.

providerBuilder.AddOtlpExporter();

var provider = providerBuilder.Build();

while (true)
{
    await Task.Delay(TimeSpan.FromSeconds(1));
    meter.CreateCounter<int>("foo").Add(1);
}
```

I have also tested this with `MeterFactory` using the `MeterOptions` and have the same issue

Aspire Dashboard:

![image](https://github.com/open-telemetry/opentelemetry-dotnet/assets/23398170/7d1ce634-8c48-4acd-ac91-0f22815df7c7)

I am expecting the `namespace` tag to show, for the `foo` counter, but it does not.

If I use `AddConsoleExporter` I get:

```
Metric Name: foo, Meter: TestMeter/1.0.0
        Meter Tags:
                namespace: jacklocaldev
(2024-06-04T11:30:20.9913140Z, 2024-06-04T11:30:59.9698360Z] LongSum
Value: 39
```

Am I misunderstanding how this API works? 

Reading [this issue](https://github.com/open-telemetry/opentelemetry-dotnet/discussions/4420) may also suggest it could just be an issue with the exporter, in my case the aspire dashboard? If that's the case I am happy to start a discussion over there.

**Comment by cijothomas** [comment, marked as answer]
If you are not seeing Meter tags in Aspire, it is better to report in that repo.
You can confirm that OTLP Exporter is working fine by using OTel Collector with its debug output.

**Reply by JackRhodes** [reply]
Thanks - I'll closed this discussion and start a discussion over there 👍
